apps/maestro/src/features/stellarHooks/useDeployAndRegisterStellarToken.ts

```diff
@@ -0,0 +1,333 @@
+        destinationChainIds,
+        gasValues: gasValues.map(value => value.toString()),
+      });
+
+      // 2. Sign the transaction
+      const { signedTxXdr } = await kit.signTransaction(transactionXDR, {
+        networkPassphrase: STELLAR_NETWORK_PASSPHRASE,
+      });
+
+      // 3. Submit the transaction
+      const rpcUrl = stellarChainConfig.rpcUrls.default.http[0];
+      const server = new rpc.Server(rpcUrl);
+
+      const tx = new Transaction(signedTxXdr, STELLAR_NETWORK_PASSPHRASE);
+
+      const initialResponse = await server.sendTransaction(tx);
+
+      if (initialResponse.status === "PENDING") {
+        onStatusUpdate?.({
+          type: "deploying",
+          txHash: initialResponse.hash,
+        });
+      }
+
+      if (
+        initialResponse.status === "ERROR" ||
+        initialResponse.status === "DUPLICATE" ||
+        initialResponse.status === "TRY_AGAIN_LATER"
+      ) {
+        const errorMessage = `Stellar transaction submission failed with status: ${initialResponse.status}. Error: ${JSON.stringify(initialResponse.errorResult)}`;
+        throw new Error(errorMessage);
+      }
+
+      const pollingResult = await pollTransaction(
+        server,
+        initialResponse.hash,
+        {
+          onStatusUpdate: (status) => {
+            if (status.type === "polling") {
+              onStatusUpdate?.({
+                type: "deploying",
+                txHash: initialResponse.hash,
+              });
+            }
+          },
+        }
+      );
+
+      if (pollingResult.status !== "SUCCESS") {
+        throw pollingResult.error;
+      }
+
+      // Use the transaction result for data extraction
+      // We need to fetch the transaction again to access resultMetaXdr
+      const getTxResponse = await server.getTransaction(initialResponse.hash);
+
+      // 5. Extract token information from the transaction response
+      let tokenId: string | undefined;
+      let tokenAddress: string | undefined;
+      let tokenManagerAddress: string | undefined;
+      let tokenManagerType: string | undefined;
+
+      // Check if we have resultMetaXdr in the response (may not be defined in the type)
+      const txResponseWithMeta = getTxResponse as any;
+      if (txResponseWithMeta.resultMetaXdr) {
+        try {
+          // Extract tokenId from the transaction return value
+          const transactionMeta = txResponseWithMeta.resultMetaXdr;
+          const returnValue = transactionMeta
+            .v3()
+            ?.sorobanMeta()
+            ?.returnValue();
+
+          if (
+            returnValue &&
+            returnValue.switch() === xdr.ScValType.scvBytes()
+          ) {
+            tokenId = returnValue.bytes().toString("hex");
+          }
+
+          // Extract tokenAddress, tokenManagerAddress, tokenManagerType, and GMP message IDs from events
+          const sorobanMeta = transactionMeta.v3()?.sorobanMeta();
+          const events = sorobanMeta?.events();
```

Comment:
i'm not sure about this but can we use [humanizeEvents](https://stellar.github.io/js-stellar-sdk/global.html#humanizeEvents) function from the stellar sdk to simplify this?